macOS/DuckDuckGo/NetworkProtection/AppTargets/BothAppTargets/VPNUIActionHandler.swift

```diff
@@ -18,32 +18,46 @@
 
 import AppLauncher
 import Foundation
+import NetworkProtection
 import NetworkProtectionIPC
 import NetworkProtectionProxy
 import NetworkProtectionUI
+import SwiftUI
 import VPNAppLauncher
+import VPNAppState
 
 /// Main App's VPN UI action handler
 ///
 final class VPNUIActionHandler {
 
     private let vpnIPCClient: VPNControllerXPCClient
     private let proxySettings: TransparentProxySettings
+    private let tunnelController: TunnelController
+    private let vpnAppState: VPNAppState
     private let vpnURLEventHandler: VPNURLEventHandler
 
     init(vpnIPCClient: VPNControllerXPCClient = .shared,
          vpnURLEventHandler: VPNURLEventHandler,
-         proxySettings: TransparentProxySettings) {
+         tunnelController: TunnelController,
+         proxySettings: TransparentProxySettings,
+         vpnAppState: VPNAppState) {
 
         self.vpnIPCClient = vpnIPCClient
         self.vpnURLEventHandler = vpnURLEventHandler
+        self.tunnelController = tunnelController
         self.proxySettings = proxySettings
+        self.vpnAppState = vpnAppState
     }
 
     func askUserToReportIssues(withDomain domain: String) async {
         let parentWindow = await WindowControllersManager.shared.lastKeyMainWindowController?.window
         await ReportSiteIssuesPresenter(userDefaults: .netP).show(withDomain: domain, in: parentWindow)
     }
+
+    @MainActor
+    private var windowControllerManager: WindowControllersManager {
+        WindowControllersManager.shared
+    }
```

Comment:
Moving the singleton here will allow us to eventually inject it and move away from referencing the singleton directly within the code.